CIP-XXXX/standard.cddl

```diff
@@ -0,0 +1,84 @@
+; Oracle Datum
+
+; The oracle datum is composed of a generic, shared and extended data.
+;   - With key 0: The generic data will be a list of different feed values.
+;   - With key 1: The shared data could not be part of the datum, but if it's
+;     present then carries the information about the timestamp and expiry of
+;     all the feeds.
+;   - With key 2: The extended data could not be part of the datum, but if it's
+;     present then carries some external information about the provider of the
+;     feeds.
+oracle_datum = #6.121(oracle_datum_list)
+oracle_datum_list =
+  [ ?  shared_data	
+  , 1* generic_data
+  , ?  extended_data
+  ]
+
+; At the moment we have two types of generic data:
+;   - price_data, for the exchange rate between two assets.
+;   - kyc_data, for “Know Your Customer” related data
+generic_data = price_data / kyc_data
+
+; A price_data entry normally defines the “Currency Pair” being priced.
+;  A “Currency Pair” is defined as “the first listed currency or base currency is
+;  bought while the second listed currency in a currency pair or quote currency is
+;  sold.” Each currency is defined separately in the entry, and referred to as the
+;  “base” and “quote” currency.
+price_data = #6.123([price_map])
+price_map =
+  { ? 0 : price        ; how many quote currency is received per base currency spent
+  , ? 1 : posixtime    ; unix timestamp related to when the price data was created
+  , ? 2 : posixtime    ; unix timestamp related to when the price data is expired
```

Comment:
Considering using a single time to denote when the value was 'observed'. In frp terminology, price is a function over 'continuous' time and it simplifies working with it in such way. So one could just have a field `? 1 : posixtime ; unix timestamp of when the price was observed`.

Any script running onchain can make their own decision on how much 'interval' to put around an observation by simply saying `contains someIntervalAroundObservation observedAt`